web/client/src/components/validation/TokenValidation.jsx

```diff
@@ -27,81 +29,83 @@ export default function ValidateToken() {
   const [credsToken, setCredsToken] = useState("");
   const [wantInfo, setWantInfo] = useState(false);
   const [info, setInfo] = useState("");
+  const [errorOpen, setErrorOpen] = useState(true);
+  const [hasError, setHasError] = useState(false);
+  const [errMessage, setErrMessage] = useState("");
 
   /**
    *
-   * @param {event} e handler for when the token info button is clicked. Will provide the message from the OAuth2l info command for the token specified.
+   * @param {param} values will take the inputted token and test it, returning if the token is valid or not.
    */
-  function getTokenInfo(e) {
-    e.preventDefault();
-    // To indicate the the info about the token is wanted
-    setWantInfo(true);
-    // Body for the request.
+  function testToken(values) {
+    // Indicated that a token was inputted and is ready to be submitted.
+    setCompleted(true);
+    // Sets the credentialsToken to be the inputted token so that it can be used in the future if user wants the information of the token.
+    setCredsToken(values["token"]);
+    // JSON body for the request.
     const requestOptions = {
-      commandtype: "info",
+      commandtype: "test",
       args: {
-        token: credsToken,
+        token: values["token"],
       },
       cachetoken: false,
       usetoken: false,
       token: "",
     };
-    // Sending the request.
-    axios
-      .post("http://localhost:8080/", requestOptions)
+
+    // Sending the Request
+    validateToken(requestOptions)
       .then(async (response) => {
-        const endofResponse = response.data.indexOf("}");
-        // Setting info that will be displayed as the OAuth2l Response from the command.
-        setInfo(response.data.substring(22, endofResponse));
+        if (response.data["OAuth2l Response"] === "0") {
+          setValid(true);
+        } else {
+          setValid(false);
+        }
+      })
+      .catch(function (error) {
+        setErrorOpen(true);
+        setHasError(true);
+        setErrMessage(error.toString());
       });
   }
 
   /**
    *
-   * @param {event} e handler for when the token reset button is clicked. Will execute the OAUth2l reset command.
+   * @param {event} e handler for when the token info button is clicked. Will provide the message from the OAuth2l info command for the token specified.
```

Comment:
This should just be the param description. Add function description to a new line